src/Psy/Command/ComposerCommand.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Psy\Command;
+
+use Composer\Console\Application;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class ComposerCommand
+ * @package Psy\Command
+ */
+class ComposerCommand extends Command
+{
+    const PIPE_WRITE = 0;
+    const PIPE_READ = 1;
+    const PIPE_ERR = 2;
+
+    /**
+     * @var array
+     */
+    protected $specification = array(
+        self::PIPE_WRITE => array("pipe", "r"),
+        self::PIPE_READ => array("pipe", "w"),
+        self::PIPE_ERR => array("pipe", "w"),
+    );
+
+    /**
+     * @var string
+     */
+    protected $composerPath;
+
+    /**
+     * @var string
+     */
+    protected $installationType;
+
+    /**
+     * @var InputInterface
+     */
+    protected $input;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('composer')
+            ->setDefinition(array(
+                new InputArgument('command-name', InputArgument::REQUIRED, ''),
+                new InputArgument('args', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, ''),
+            ))
+            ->setDescription('Composer installation.')
+            ->setHelp(
+                <<<HELP
+composer library Installs the library
+
+if composer is not found will install it locally
+HELP
+            );
+        $this->ignoreValidationErrors();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+
+        if (!$this->checkComposerInstallation()) {
+            $this->installComposer();
+        }
+
+        $this->callComposerBootstrap();
+
+        // extract real command name
+        $tokens = preg_split('{\s+}', $input->__toString());
```

Comment:
For the other command i am not sure if it doesn't have downsides, but for this command if i let the engine parse the options we will end with composer calls that have missing some arguments, for example, if the application parses the options and arguments, you will not be able to do a call like this for example : `composer search foo --only-names`